evm/spec/tables/byte-packing.tex

```diff
@@ -1,4 +1,54 @@
 \subsection{Byte Packing}
 \label{byte-packing}
 
-TODO
+The BytePacking STARK module is used for reading and writing to memory non-empty byte sequences of length at most 32.
+The "packing" term highlights that reading a sequence in memory will pack the bytes into an EVM word (i.e. U256), while
+the "unpacking" operation consists in breaking down an EVM word into its byte sequence and writing it in memory.
+
+This allows faster memory copies between two memory locations, as well as faster memory reset
+(see \href{https://github.com/0xPolygonZero/plonky2/blob/main/evm/src/cpu/kernel/asm/memory/memcpy.asm}{memcpy.asm} and 
+\href{https://github.com/0xPolygonZero/plonky2/blob/main/evm/src/cpu/kernel/asm/memory/memset.asm}{memset.asm} modules).
+
+The `BytePackingStark' table will have $\ell$ rows per packing/unpacking operation, where $0 < \ell \leq 32$ is the length of the sequence being processed.
+
+Each row contains the following columns:
+\begin{enumerate}
+    \item 5 columns containing information on the initial memory address from which the sequence starts
+    (namely a flag differentiating read and write operations, address context, segment and offset values, as well as timestamp),
+    \item $f_{\texttt{end}}$, a flag indicating the end of a sequence,
+    \item 32 columns $b_i$ indicating the index of the byte being read/written at a given row,
+    \item 32 columns $v_i$ indicating the values of the bytes that have been read or written during a sequence,
+    \item 2 columns $r_i$ needed for range-checking the the byte values.
+\end{enumerate}
+
+\paragraph{Notes on columns generation:}
+Whenever a byte unpacking operation is called, the value $\texttt{val}$ is read from the stack. But because there is a discrepancy in the endianness in CpuStark and MemoryStark, we need to convert $\texttt{val}$ to a byte sequence in Little-Endian. Only then do we resize it to the appropriate length, thus pruning extra zeros and higher bytes in the process. Finally, we reverse the byte order and write this new sequence into the $v_i$ columns of the table. 
```

Comment:
```suggestion
Whenever a byte unpacking operation is called, the value $\texttt{val}$ is read from the stack, but because the CpuStark and MemoryStark use different endianness, we need to convert $\texttt{val}$ to a little-endian byte sequence. Only then do we resize it to the appropriate length, and prune extra zeros and higher bytes in the process. Finally, we reverse the byte order and write this new sequence into the $v_i$ columns of the table. 
```

Reply:
For my education: why do they use different endianness?

Reply:
Sorry, what I wrote was wrong actually: the different endianness is between the EVM and the STARKs (I am not sure why exactly, but I guess it's because it's easier to code in little-endian and we need big-endian for some modules such as Keccak). I changed this in the rest of the file as well